sys/time.c

```diff
@@ -14,24 +14,58 @@
  */
 
 #include <sys/time.h>
+#include <sys/msg.h>
+#include <sys/file.h>
 #include <errno.h>
 #include <stdlib.h>
 
 #include "posix/utils.h"
 
-extern int sys_utimes(const char *filename, const struct timeval times[2]);
+
+extern int sys_futimes(int fd, const struct timeval times[2]);
+
+
+/* path needs to be canonical */
+static int utimes_internal(const char *path, const struct timeval times[2])
+{
+	oid_t oid, dev;
+	msg_t msg = { 0 };
+	struct timeval now;
+	int err;
+
+	if (lookup(path, &oid, &dev) < 0)
+		return -ENOENT;
+
+	msg.type = mtSetAttr;
+	msg.i.attr.oid = oid;
+
+	if (times == NULL)
+		gettimeofday(&now, NULL);
+
+	msg.i.attr.type = atMTime;
+	msg.i.attr.val = times == NULL ? now.tv_sec : times[1].tv_sec;
+	if (((err = msgSend(oid.port, &msg)) < 0) || ((err = msg.o.attr.err) < 0))
```

Comment:
Assignment inside of if condition is against MISRA rule, we are abandoning this code idiom.

Reply:
hey, do You know if this rule also applies to assignments inside `while` conditions? (should probably apply also, so we need to rewrite our `-EINTR` loops in libphoenix?)

Reply:
Yeah, I do, missed that one during second reading - to be fixed too.

We need to fix all occurrences, for now during natural code evolution, later as part of making static code analysis pass without errors ;)